webapp/src/components/setting.jsx

```diff
@@ -20,24 +21,27 @@ export default class Setting extends React.PureComponent {
             inputId,
             label,
             required,
+            hideRequiredStar,
         } = this.props;
 
         return (
-            <div className='form-group'>
+            <div className='form-group less'>
+                {label &&
                 <label
-                    className='control-label'
+                    className='control-label margin-bottom x2'
                     htmlFor={inputId}
                 >
                     {label}
                 </label>
-                {required && (
-                    <span
-                        className='error-text'
-                        style={{marginLeft: '3px'}}
-                    >
-                        {'*'}
-                    </span>
-                )}
+                }
+                {required && !hideRequiredStar &&
```

Comment:
For readability. And add the closing paren

```suggestion
                {required && !hideRequiredStar && (
```